"""2. Создайте собственный класс-исключение, обрабатывающий ситуацию деления на ноль.
Проверьте его работу на данных, вводимых пользователем. При вводе нуля в качестве делителя
программа должна корректно обработать эту ситуацию и не завершиться с ошибкой."""


class MyErr(Exception):
    def __init__(self, txt):
        self.txt = txt


num_1 = int(input('Введите число которое надо разделить: '))
num_2 = int(input('делитель: '))

try:
    if num_2 == 0:
        raise MyErr('not null')
except MyErr as err:
    print(err)
else:
    print(num_1 / num_2)
